### PR TITLE
fix: compact_map get error mismatching cokie

### PR DESCRIPTION
**Pull request by binbinshi**
# What problem are we solving?

when use snowflak gen id ，The following error occurs :
write to local disk: mismatching cookie xxxxxxx
the file write failue

# How are we solving the problem?

fix the bug of compact_map get error ; when the slot end small than the key return  nil false

# How is the PR tested?

add compact get test 
test idx  
[1655451851227025229-502m-single-1_388.idx.zip](https://github.com/seaweedfs/seaweedfs/files/9654785/1655451851227025229-502m-single-1_388.idx.zip)

# Checks
- [x] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by chrislusf** [comment]
tests failed.

File: weed/storage/needle_map/compact_map_perf_test.go [hunk 1]
```diff
@@ -90,3 +90,34 @@ func PrintMemUsage(totalRowCount uint64) {
 func bToMb(b uint64) uint64 {
 	return b / 1024 / 1024
 }
+
+func TestCompactSection_Get(t *testing.T) {
```
Comment (chrislusf) [inline weed/storage/needle_map/compact_map_perf_test.go]: move to compact_map_test.go

File: weed/storage/needle_map/compact_map_perf_test.go [hunk 2]
```diff
@@ -90,3 +90,34 @@ func PrintMemUsage(totalRowCount uint64) {
...
+	var maps []*CompactMap
+	totalRowCount := uint64(0)
+	indexFile, ie := os.OpenFile("../../1655451851227025229-502m-single-1_388.idx",
```
Comment (chrislusf) [inline weed/storage/needle_map/compact_map_perf_test.go]: where is this file? Seems this test does not work at all.

Comment (binbinshi) [reply weed/storage/needle_map/compact_map_perf_test.go]: [1655451851227025229-502m-single-1_388.idx.zip](https://github.com/seaweedfs/seaweedfs/files/9654785/1655451851227025229-502m-single-1_388.idx.zip)

Comment (binbinshi) [reply weed/storage/needle_map/compact_map_perf_test.go]: the file  is 38MB, so i didn't commit it

File: weed/storage/needle_map/compact_map.go [hunk 3]
```diff
@@ -144,9 +144,13 @@ func (cs *CompactSection) deleteOverflowEntry(key SectionalNeedleId) {
 
 // return old entry size
 func (cs *CompactSection) Delete(key NeedleId) Size {
-	skey := SectionalNeedleId(key - cs.start)
 	cs.Lock()
+	defer cs.RUnlock()
```
Comment (chrislusf) [inline weed/storage/needle_map/compact_map.go]: this is wrong. The test did not cover this function?

File: weed/storage/needle_map/compact_map_perf_test.go (hunk 2 above)
Comment (chrislusf) [reply weed/storage/needle_map/compact_map_perf_test.go]: try to create a test case directly. reproduce first, then fix.

Comment (binbinshi) [reply weed/storage/needle_map/compact_map_perf_test.go]: reproduce  pipline 
1. remove the compact_map.go Get function  line 169-171
2. run the test TestCompactSection_Get func ,  may test failed of  get error 
3. recovery the compact_map.go Get function  line 169-171  , run the TestCompactSection_Get test will pass

**Pull request merged**